activesupport/test/logger_test.rb

```diff
@@ -258,6 +258,50 @@ def test_logger_level_local_thread_safety
     assert_level(Logger::INFO)
   end
 
+  def test_logger_level_main_fiber_safety
+    @logger.level = Logger::INFO
+    assert_level(Logger::INFO)
+
+    fiber = Fiber.new do
+      assert_level(Logger::INFO)
+    end
+
+    @logger.silence(Logger::ERROR) do
+      assert_level(Logger::ERROR)
+      fiber.resume
+    end
+  end
+
+  def test_logger_level_local_fiber_safety
+    @logger.level = Logger::INFO
+    assert_level(Logger::INFO)
+
+    another_fiber = Fiber.new do
+      @logger.silence(Logger::ERROR) do
+        assert_level(Logger::ERROR)
+        @logger.silence(Logger::DEBUG) do
+          assert_level(Logger::DEBUG)
+        end
+      end
+
+      assert_level(Logger::INFO)
+    end
+
+    Fiber.new do
+      @logger.silence(Logger::ERROR) do
+        assert_level(Logger::ERROR)
+        @logger.silence(Logger::DEBUG) do
+          another_fiber.resume
+          assert_level(Logger::DEBUG)
+        end
+      end
+
+      assert_level(Logger::INFO)
+    end.resume
+
+    assert_level(Logger::INFO)
+  end
```

Comment:
We could also split and rename test_logger_level_local_thread_safety to
test_logger_level_local_thread_safety
test_logger_level_local_fiber_safety
to explicitly show that we test now two different contexts